desktop/components/react/utils/template_renderer.js

```diff
@@ -20,22 +28,26 @@ export function makeTemplate (templates, options = {}) {
     ? templates
     : [templates]
 
-  const basePath = options.basePath || ''
+  const {
+    basePath,
+    compilerOptions,
+    locals
+  } = merge(defaultOptions, opts)
 
   const templateFns = templates.map(file => {
     return jade.compileFile(path.join(basePath, file), {
       cache: true,
-      ...options
+      ...compilerOptions
     })
   })
 
-  return {
-    render: (locals = {}) => {
-      const renderTemplate = locals => template => template(locals)
+  const renderTemplate = locals => template => template(locals)
```

Comment:
(Reading back through this to try to wrap my head around the cache bug)

Since we're just about to apply this function to `locals`, I don't think we want to make a function that takes `locals` as a parameter. Would be simpler to just do:

```
const renderTemplate = template => template(locals)
const renderedTemplates = templateFns.map(renderTemplate)
```

But I'd probably avoid naming the lambda altogether and just do

```
const renderedTemplates = templateFns.map(template => template(locals))
```

Incidentally, in Scala (or another language with anonymous lambda parameters), you'd do `templateFns.map(_(locals))` or maybe with a little more explicitness `templateFns.map(_.apply(locals))`.